Yoast/YoastSEO.js
# Remove TypeScript

## Summary

This PR can be summarized in the following changelog entry:

* _Drop TypeScript support_

## Relevant technical choices:

It has given us no gains and a bit of annoyance. We can add it later,
but that needs to occur with an effort to rewrite at least a great part
of the codebase to use types. That way we actually get some results from
using TypeScript.

## Test instructions

This PR can be tested by following these steps:

* Everything should still build.
* `yarn test` should pass.
* `grunt check` should pass.
* No traces of TypeScript should be left.
